akka-projection-dynamodb/src/main/scala/akka/projection/dynamodb/internal/DynamoDBProjectionImpl.scala

```diff
@@ -202,14 +203,100 @@ private[projection] object DynamoDBProjectionImpl {
             case Duplicate =>
               FutureDone
             case RejectedSeqNr =>
-              triggerReplayIfPossible(sourceProvider, offsetStore, envelope).map(_ => Done)(ExecutionContext.parasitic)
+              replayIfPossible(envelope).map {
+                case true  => Done
+                case false => throwRejectedEnvelopeAfterFailedReplay("query", sourceProvider, envelope)
+              }(ExecutionContext.parasitic)
```

Comment:
Too much failing now? Always fails for queries if not replayed. Can move the throw to just for expected replay count.

Reply:
Fixed, only throwing for the incorrect count.